integration/dashboards_test.go

```diff
@@ -151,66 +151,65 @@ func cleanUpDashboard(t *testing.T, id int) {
 	graphs := []datadog.Graph{}
 	graphs = append(graphs, graph)
 	return graphs
 }
 
 func createAdvancedTimeseriesGraph() []datadog.Graph {
 	graphDefinition := datadog.Graph{}.Definition
-	graphDefinition.Viz = "timeseries"
+	graphDefinition.Viz = datadog.String("timeseries")
 	r := datadog.Graph{}.Definition.Requests
-	pallette := "warm"
 
 	graphDefinition.Requests = append(r, datadog.GraphDefinitionRequest{
-		Query:   "avg:system.mem.free{*}",
-		Stacked: false,
-		Type:    "bars",
-		Style:   &datadog.GraphDefinitionRequestStyle{Palette: &pallette},
+		Query:   datadog.String("avg:system.mem.free{*}"),
+		Stacked: datadog.Bool(false),
+		Type:    datadog.String("bars"),
+		Style:   &datadog.GraphDefinitionRequestStyle{Palette: datadog.String("warm")},
 	})
-	graph := datadog.Graph{Title: "Custom type and style graph", Definition: graphDefinition}
+	graph := datadog.Graph{Title: datadog.String("Custom type and style graph"), Definition: graphDefinition}
 	graphs := []datadog.Graph{}
 	graphs = append(graphs, graph)
 	return graphs
 }
 
 func createCustomGraph() []datadog.Graph {
 	graphDefinition := datadog.Graph{}.Definition
-	graphDefinition.Viz = "query_value"
+	graphDefinition.Viz = datadog.String("query_value")
 	r := datadog.Graph{}.Definition.Requests
 	graphDefinition.Requests = append(r, datadog.GraphDefinitionRequest{
-		Query:      "( sum:system.mem.used{*} / sum:system.mem.free{*} ) * 100",
-		Stacked:    false,
-		Aggregator: "avg",
+		Query:      datadog.String("( sum:system.mem.used{*} / sum:system.mem.free{*} ) * 100"),
+		Stacked:    datadog.Bool(false),
+		Aggregator: datadog.String("avg"),
 		ConditionalFormats: []datadog.DashboardConditionalFormat{
 			{
-				Comparator: ">",
-				Value:      json.Number("99.9"),
-				Palette:    "white_on_green"},
+				Comparator: datadog.String(">"),
+				Value:      datadog.JsonNumber(json.Number("99.9")),
+				Palette:    datadog.String("white_on_green")},
 			{
-				Comparator: ">=",
-				Value:      json.Number("99"),
-				Palette:    "white_on_yellow"},
+				Comparator: datadog.String(">="),
+				Value:      datadog.JsonNumber(json.Number("99")),
+				Palette:    datadog.String("white_on_yellow")},
 			{
-				Comparator: "<",
-				Value:      json.Number("99"),
-				Palette:    "white_on_red"}}})
-	graph := datadog.Graph{Title: "Mandatory graph 2", Definition: graphDefinition}
+				Comparator: datadog.String("<"),
+				Value:      datadog.JsonNumber(json.Number("99")),
+				Palette:    datadog.String("white_on_red")}}})
+	graph := datadog.Graph{Title: datadog.String("Mandatory graph 2"), Definition: graphDefinition}
 	graphs := []datadog.Graph{}
 	graphs = append(graphs, graph)
 	return graphs
 }
 
 func assertDashboardEquals(t *testing.T, actual, expected *datadog.Dashboard) {
-	if actual.Title != expected.Title {
-		t.Errorf("Dashboard title does not match: %s != %s", actual.Title, expected.Title)
+	if *actual.Title != *expected.Title {
```

Comment:
Makes sense. Well, I'm agreed on using the methods like `Int` and `String` etc. Although if we do go with the `json.Number` route we should make it `Number` instead? (Oh ugh, nevermind, I forget Go can't overload arguments... oh wait, I guess we could make it take an `interface{}` and unbox it ourselves??)